modules/openapi/src/internals/OpenApiToIModel.scala

```diff
@@ -252,6 +252,15 @@ private[openapi] class OpenApiToIModel[F[_]: Parallel: TellShape: TellError](
         val code = output.code
         code >= 200 && code < 300
     }.toList match {
+      case output :: Nil if output.code == 204 =>
```

Comment:
Even though this works to solve the raised issue, it's actually not fixing the underlying bug : the fact that the status code is 204 should be inconsequential to the logic : you should be able to replace `204` with `200` in the test input and still get a successful result, even if it doesn't quite make sense from a REST perspective.

Reply:
Following the code, I think the problem lies in these lines : 

https://github.com/disneystreaming/smithy-translate/blob/b7bf1ed7a3992899e731d924c39f15c7c14fc169/modules/openapi/src/internals/OpenApiToIModel.scala#L333-L337

When the fields are empty, a structure should probably still be recorded, but we should have a [post-processor](https://github.com/disneystreaming/smithy-translate/tree/main/modules/compiler-core/src/internals/postprocess) that'd turn all empty-structures that don't have hints to `Unit`